### ISSUE TITLE
don't override Symbolics.jl sharding heuristics

### ISSUE THREAD

**Issue by wsphillips** [SciML/ModelingToolkit.jl#1782]
When parallel form is not specified, Symbolics.jl will automatically shard large systems:

https://github.com/JuliaSymbolics/Symbolics.jl/blob/06cdbf6dbee7af3a102d1941639bb820177be916/src/build_function.jl#L261-L263

...and otherwise falls back to just `SerialForm`:

https://github.com/JuliaSymbolics/Symbolics.jl/blob/06cdbf6dbee7af3a102d1941639bb820177be916/src/build_function.jl#L302-L309

Here in MTK the default is `parallel = SerialForm()`: 

https://github.com/SciML/ModelingToolkit.jl/blob/d417a2cf5158537905eb93d6a3aa8d6e44bd9888/src/systems/diffeqs/abstractodesystem.jl#L524-L530

...which causes automatic sharding to be skipped in `Symbolics.build_function`. This will break large systems which could otherwise run (even if slowly) in single-thread sharded form.

tl;dr we should probably set `parallel = nothing` in `process_DEProblem` to match `Symbolics.jl` assumptions and heuristics.
